codes/models/modules/architectures/DVDNet_arch.py

```diff
@@ -0,0 +1,68 @@
+import torch
+import torch.nn as nn
+
+def vertcal_upscale(x, upfield=True):
+
+    n, c, h, w = x.shape
+    h *= 2
+    t = x
+    if upfield:
+        out = torch.cat([t, torch.zeros_like(t)], 3)
+    else:
+        out = torch.cat([torch.zeros_like(t), t], 3)
+    out = torch.reshape(out, (n, c, h, w))
+    return out
+
+
+def replace_field(x, input_image, upfield=True):
+
+    upper_input = input_image[:, :, 0::2, :]
+    lower_input = input_image[:, :, 1::2, :]

```

Comment:
If H is an even value, would lower_input be missing one line?

Reply:
Yes, but aimilar to the other issue I do not think this will be a realistic issue as interlaced frames will always have an even height